src/posix.c

```diff
@@ -239,9 +239,13 @@ int p_mmap(git_map *out, size_t len, int prot, int flags, int fd, git_off_t offs
 	out->data = NULL;
 	out->len = 0;
 
-	if ((prot & GIT_PROT_WRITE) && ((flags & GIT_MAP_TYPE) == GIT_MAP_SHARED)) {
-		giterr_set(GITERR_OS, "Trying to map shared-writeable");
-		return -1;
+	if (prot & GIT_PROT_WRITE) {
+		if ((p_lseek(fd, offset, SEEK_SET) < 0) || p_write(fd, out->data, len)) {
```

Comment:
What is the `p_write()` call supposed to do? `out->data` is `NULL` and `len` is `0` since we just set them. What would this do besides be a no-op?